dataeng/resources/opsgenie-enable-heartbeat.sh

```diff
@@ -0,0 +1,31 @@
+# Enables an Opsgenie Heartbeat, while also setting the expected duration.
+#
+# Assumes that the heartbeat is already created in the Opsgenie settings here:
+# https://edx.app.opsgenie.com/settings/heartbeat
+#
+# Required environment variables:
+# OPSGENIE_HEARTBEAT_NAME: Name of the already-created Opsgenie heartbeat.
+# OPSGENIE_HEARTBEAT_CONFIG_KEY: API key provided by Opsgenie which is authorized to modify heartbeat configuration.
+# OPSGENIE_HEARTBEAT_DURATION_NUM: Specifies how often a heartbeat message should be expected.
+# OPSGENIE_HEARTBEAT_DURATION_UNIT: Interval specified as minutes, hours or days.
+
+
+OPSGENIE_HEARTBEAT_API_URL="https://api.opsgenie.com/v2/heartbeats"
+
+if [ -n "$OPSGENIE_HEARTBEAT_NAME" ] && \
```

Comment:
Actually one question- if this script is getting called, shouldn't all of these exist? If they don't should we throw an error instead of ignoring it?

Reply:
Look at the proposed job config for answer-distribution, we don't always configure these variables for every environment.  Ideally I think we would switch on these at a higher level (i.e. the job DSL level), like we do with setting the cron trigger, but I'm okay with either solution.

Reply:
I guess my guiding principle here was that this heartbeat monitoring shouldn't interfere with the job triggering on error. Because of that, I also chose *not* to check the `curl` return codes either. A loud failure on a misconfiguration would arguably be better than silently ignoring that condition, as the code is currently written. But as we're doing this stuck-job checking for a limited number of jobs - and we're not on jenkins-new here - I'm going forward with this for now.